### PR TITLE
ROC-4322 calculate last payment date

### PR DESCRIPTION
**Pull request by rattur**
### JIRA link

https://tools.hmcts.net/jira/browse/ROC-4322

### Change description

Refactor the FinalPaymentDateCalculator class to return a Decision Type 

### Work checklist

- [ x] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [ ] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x ] No

### CODE REVIEW

**Comment by varma02** [comment]
you can perhaps add test to check what happens when defendantPaymentDate, claimantPaymentDate, courtGeneratedPaymentDate are same!! and all other combinations.

I think this is a scenario that can happen.

File: src/test/app/common/court-calculations/courtDetermination.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    })
+
+    it('should return a claimant or defendant decision type when claimantPaymentDate and defendantPaymentDate are the same', () => {
+
+      let defendantPaymentDate = new LocalDate(2019, 12, 31).toMoment()
+      let claimantPaymentDate = new LocalDate(2019, 12, 31).toMoment()
+      let courtGeneratedPaymentDate = new LocalDate(2019, 10, 11).toMoment()
+
+      expect(CourtDetermination.calculateDecision(defendantPaymentDate, claimantPaymentDate, courtGeneratedPaymentDate)).to.equal(DecisionType.CLAIMANT)
+    })
```
Comment (varma02) [inline src/test/app/common/court-calculations/courtDetermination.ts]: Since its valid to return either Claimant or Defendant decision type, you could probably try something like this:

`expect(Object.values(DecisionType).includes(CourtDetermination.calculateDecision(defendantPaymentDate, claimantPaymentDate, courtGeneratedPaymentDate)))`

File: src/main/app/common/court-calculations/courtDetermination.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  COURT = 'COURT'
+}
+
+export class CourtDetermination {
+
+  static calculateDecision (defendantPaymentDate: Moment,
+                             claimantPaymentDate: Moment,
+                             courtGeneratedPaymentDate: Moment): DecisionType {
+
+    if (claimantPaymentDate.isAfter(defendantPaymentDate) || claimantPaymentDate.isAfter(courtGeneratedPaymentDate)) {
```
Comment (varma02) [inline src/main/app/common/court-calculations/courtDetermination.ts]: could you please try isSameOrAfter instead of isAfter as dates can be same and run all tests

File: src/main/app/common/court-calculations/courtDetermination.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      return DecisionType.CLAIMANT
+    }
+    if (claimantPaymentDate.isBefore(courtGeneratedPaymentDate)) {
+      if (defendantPaymentDate.isBefore(courtGeneratedPaymentDate)) {
```
Comment (varma02) [inline src/main/app/common/court-calculations/courtDetermination.ts]: same here, could you please try isSameOrBefore instead of isAfter as dates can be same and run all tests

File: src/main/app/common/court-calculations/courtDetermination.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  DEFENDANT = 'DEFENDANT',
+  COURT = 'COURT'
+}
+
+export class CourtDetermination {
+
+  static calculateDecision (defendantPaymentDate: Moment,
+                             claimantPaymentDate: Moment,
+                             courtGeneratedPaymentDate: Moment): DecisionType {
+
```
[the remark this answers is not in the document]
Comment (rattur) [reply src/main/app/common/court-calculations/courtDetermination.ts]: Done

File: src/main/app/common/court-calculations/courtDetermination.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    if (defendantPaymentDate === undefined || claimantPaymentDate === undefined || courtGeneratedPaymentDate === undefined) {
```
Comment (sabahirfan) [inline src/main/app/common/court-calculations/courtDetermination.ts]: `if (!defendantPaymentDate || !claimantPaymentDate || !courtGeneratedPaymentDate)`

File: src/test/app/common/court-calculations/courtDetermination.ts [hunk 6]
```diff
@@ -0,0 +1,70 @@
+import { expect } from 'chai'
+
+import { CourtDetermination, DecisionType } from 'common/court-calculations/courtDetermination'
+import moment = require('moment')
+
+describe('CourtDetermination', () => {
+
+  context('calculateDecision', () => {
+
+    it('should throw an error if defendantPaymentDate, claimantPaymentDate or courtGeneratedPaymentDate are undefined', () => {
```
Comment (sabahirfan) [inline src/test/app/common/court-calculations/courtDetermination.ts]: add test for `null` params
